smart_contracts/contract/src/contract_api/runtime.rs

```diff
@@ -187,6 +187,37 @@ pub fn get_named_arg<T: FromBytes>(name: &str) -> T {
     bytesrepr::deserialize(arg_bytes).unwrap_or_revert_with(ApiError::InvalidArgument)
 }
 
+/// Returns given named argument passed to the host for the current module invocation.
+/// If the argument is not found, returns `None`.
+///
+/// Note that this is only relevant to contracts stored on-chain since a contract deployed directly
+/// is not invoked with any arguments.
+pub fn try_get_named_arg<T: FromBytes>(name: &str) -> Option<T> {
+    let arg_size = get_named_arg_size(name)?;
+    let arg_bytes = if arg_size > 0 {
+        let res = {
+            let data_non_null_ptr = contract_api::alloc_bytes(arg_size);
+            let ret = unsafe {
+                ext_ffi::casper_get_named_arg(
+                    name.as_bytes().as_ptr(),
+                    name.len(),
+                    data_non_null_ptr.as_ptr(),
+                    arg_size,
+                )
+            };
+            let data =
+                unsafe { Vec::from_raw_parts(data_non_null_ptr.as_ptr(), arg_size, arg_size) };
+            api_error::result_from(ret).map(|_| data)
+        };
+        // Assumed to be safe as `get_named_arg_size` checks the argument already
+        res.unwrap_or_revert()
+    } else {
+        // Avoids allocation with 0 bytes and a call to get_named_arg
+        Vec::new()
+    };
+    bytesrepr::deserialize(arg_bytes).unwrap_or_revert_with(ApiError::InvalidArgument)
```

Comment:
@moubctez Hi, something is strange here I don't get this function to work as I understood its purpose, not sure if this is on my contract but I get reverted with InvalidArgument passing value, according to the comment (not an Option though)

I would invite to test this function again because according to my test get_named_arg and  try_get_named_arg do the exact same thing with Option<T> for get_named_arg, while try_get_named_arg should handle absence of argument (not None) as per code comment.
test calling
```
        let _ = args.insert("TEST1", None::<u64>);
        let _ = args.insert("TEST2", None::<u64>);
        let _ = args.insert("TEST3", 111_u64);
        let _ = args.insert("TEST4", Some(111_u64));
        let _ = args.insert("TEST5", 111_u64);
```    
       
 contract     
```
    let test1: Option<u64> = get_named_arg("TEST1");
    let test2: Option<u64> = try_get_named_arg("TEST2");

    let test3: u64 = get_named_arg("TEST3");
    let test4: Option<u64> = try_get_named_arg("TEST4");
    
    runtime::print(&format!("{:?}", test1));
    runtime::print(&format!("{:?}", test2));
    runtime::print(&format!("{:?}", test3));
    runtime::print(&format!("{:?}", test4));

    let test5: Option<u64> = try_get_named_arg("TEST5");
    runtime::print(&format!("will revert, never printed{:?}", test5));
        
```

Regards

Reply:
I've re-examined the code and as far as I can recall this is a leftover from my attempt to handle optional arguments that is not used anywhere and can be removed.

Reply:
Hi, I believe `try_get_named_arg` intention was to handle no-value/value but now it deals with no-value/Some(value) which is not really expected on usage. get_named_arg deals with None/Some(value) which requires the parameter to be given has Option<T> in any case. 

There is however a use case thus for an get_optional_named_arg not expecting an Option<T> but I think right now it's indeed better to remove this function as confusing.

 https://github.com/casper-network/casper-node/pull/4786